routers/repo/blame.go

```diff
@@ -0,0 +1,231 @@
+package repo
+
+import (
+	"bytes"
+	"fmt"
+	gotemplate "html/template"
+	"strings"
+
+	"code.gitea.io/gitea/models"
+
+	"code.gitea.io/git"
+	"code.gitea.io/gitea/modules/base"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/highlight"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/markup"
+)
+
+const (
+	tplBlame base.TplName = "repo/home"
+)
+
+// RefBlame render blame page
+func RefBlame(ctx *context.Context) {
+
+	fileName := ctx.Repo.TreePath
+	if len(fileName) == 0 {
+		ctx.NotFound("Blame FileName", nil)
+		return
+	}
+
```

Comment:
You need to check whether the file is in LFS or not. If it's in LFS you probably should bail. (What does GitHub do for LFS files?)

Reply:
GitHub seems to support that: https://github.com/jereksel/lfs-test